processor/unrollprocessor/processor.go

```diff
@@ -0,0 +1,114 @@
+package unrollprocessor
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+)
+
+type unrollProcessor struct {
+	cfg *Config
+}
+
+// newUnrollProcessor returns a new unrollProcessor.
+func newUnrollProcessor(config *Config) (*unrollProcessor, error) {
+	if err := config.Validate(); err != nil {
+		return nil, fmt.Errorf("invalid configuration: %w", err)
+	}
+
+	return &unrollProcessor{
+		cfg: config,
+	}, nil
+}
+
+// ProcessLogs implements the processor interface
+func (p *unrollProcessor) ProcessLogs(_ context.Context, ld plog.Logs) (plog.Logs, error) {
+	newLogs := plog.NewLogs()
+	resourceLogs := ld.ResourceLogs()
+	var errs error
+	for i := 0; i < resourceLogs.Len(); i++ {
+		resourceLog := resourceLogs.At(i)
+		sl := resourceLog.ScopeLogs()
+		p.siftScopeLogs(sl)
+	}
+	resourceLogs.CopyTo(newLogs.ResourceLogs())
+	return newLogs, errs
+}
+
+// siftScopeLogs will iterate through scope logs and copy over new logs from siftLogRecords
+func (p *unrollProcessor) siftScopeLogs(sl plog.ScopeLogsSlice) {
+	for i := 0; i < sl.Len(); i++ {
+		scopeLogs := sl.At(i)
+		l := scopeLogs.LogRecords()
+		p.siftLogRecords(l).CopyTo(scopeLogs.LogRecords())
```

Comment:
Same - if we're not unrolling anything on the scope, we shouldn't need to copy or change it in any way. Just drill in and work on the `LogRecordSlice` it contains.